src/test/38_nlb/test.xprocspec.xsl

```diff
@@ -0,0 +1,177 @@
+									</li>
+								</xsl:if>
+								<xsl:variable name="result" select="x:expect[preceding-sibling::x:context[1]/x:document[@port='result']]/x:document"/>
+								<xsl:if test="$result">
+									<li>
+										<xsl:call-template name="x:document-as-code">
+											<xsl:with-param name="x:document" select="$result"/>
+											<xsl:with-param name="with-title">result</xsl:with-param>
+											<xsl:with-param name="with-class" select="if ($result/parent::*/@step='x:pef-compare') then 'pef' else ''"/>
+										</xsl:call-template>
+									</li>
+								</xsl:if>
+							</ul>
+						</wrap>
+					</xsl:variable>
+					<xsl:apply-templates select="$ul/*/html:ul"/>
+				</x:documentation>
+			</xsl:if>
+			<xsl:apply-templates select="node()"/>
+		</xsl:copy>
+	</xsl:template>
+	
+	<xsl:template name="x:document-as-code">
+		<xsl:param name="x:document"/>
+		<xsl:param name="with-title"/>
+		<xsl:param name="with-class"/>
+		<xsl:choose>
+			<xsl:when test="$x:document/@type='file'">
+				<a href="{$x:document/@href}" class="{string-join(('code', $with-class), ' ')}">
+					<xsl:sequence select="$with-title"/>
+				</a>
+			</xsl:when>
+			<xsl:when test="$x:document/@type='inline'">
+				<span>
+					<xsl:sequence select="$with-title"/>
+					<div class="{string-join(('document', $with-class), ' ')}">
+						<code>
+							<xsl:if test="$with-class">
+								<xsl:attribute name="class" select="$with-class"/>
+							</xsl:if>
+							<xsl:apply-templates select="$x:document/*" mode="serialize"/>
+						</code>
+						<xsl:if test="tokenize($with-class, '\s+')='pef'">
+							<xsl:sequence select="$x:document/pef:pef"/>
+							<a class="pef-xml" href="#">xml</a>
+							<a class="pef-unicode" href="#">unicode</a>
+							<a class="pef-ascii" href="#">ascii</a>
+						</xsl:if>
+					</div>
+				</span>
+			</xsl:when>
+		</xsl:choose>
+	</xsl:template>
+	
+	<xsl:template match="html:a[tokenize(@class, '\s+')='code']">
+		<span>
+			<xsl:copy>
+				<xsl:apply-templates select="@*|node()"/>
+			</xsl:copy>
+			<xsl:variable name="class" select="string-join(tokenize(@class, '\s+')[not('code'=.)], ' ')"/>
+			<div class="{string-join(('document', $class), ' ')}">
+				<code>
+					<xsl:if test="$class!=''">
+						<xsl:attribute name="class" select="$class"/>
+					</xsl:if>
+					<xsl:value-of select="unparsed-text(resolve-uri(@href, base-uri(/*)))"/>
+				</code>
+				<xsl:if test="tokenize($class, '\s+')='pef'">
+					<xsl:apply-templates select="document(resolve-uri(@href, base-uri(/*)))/pef:pef"/>
+					<a class="pef-xml" href="#">xml</a>
+					<a class="pef-unicode" href="#">unicode</a>
+					<a class="pef-ascii" href="#">ascii</a>
+				</xsl:if>
+			</div>
+		</span>
+	</xsl:template>
+	
+	<xsl:template match="pef:page">
+		<xsl:variable name="rows" select="xs:integer(number(ancestor::*[@rows][1]/@rows))"/>
+		<xsl:variable name="cols" select="xs:integer(number(ancestor::*[@cols][1]/@cols))"/>
+		<!--
```

Comment:
By the way which Unicode braille <-> ASCII mapping should I use to display Norwegian braille? I believe NLB also uses the Easy Embosser Utility? There should be a similar option in the user interface of EEU.